front/lib/amplitude/node/index.ts

```diff
@@ -185,6 +186,38 @@ export function trackDataSourceCreated(
   );
 }
 
+export function trackDataSourceUpdated(
+  auth: Authenticator,
+  {
+    dataSource,
+  }: {
+    dataSource: DataSourceType;
+  }
+) {
+  const userId = auth.user()?.id;
+  const workspace = auth.workspace();
+  if (!workspace || !userId) {
+    return;
+  }
+  const amplitude = getBackendClient();
+  const event = new DataSourceUpdated({
+    dataSourceName: dataSource.name,
+    dataSourceProvider: dataSource.connectorProvider || "",
+    workspaceName: workspace.name,
+    workspaceId: workspace.sId,
+    assistantDefaultSelected: dataSource.assistantDefaultSelected,
+  });
+
+  amplitude.track(
+    `user-${userId}`,
+    { ...event, groups: { [GROUP_TYPE]: workspace.sId } },
+    {
+      time: Date.now(),
+      insert_id: `data_source_updated_${dataSource.id}_${Date.now()}`,
```

Comment:
The current timestamp is not good enough. This means there is no dedup key, in which case it would be better to just not set one. Because calling this function twice for the same "event" would result in tracking this event twice on Amplitude, so lets be explicit about it and not have a dedup key.

The concept of a dedup key here is a bit strange since there is no record in the DB about all the versions of the data source, so I a ok-ish not providing it.

Reply:
I did not want to update DataSourceType to expose updatedAt just for tracking :/

Reply:
so lets just remove the insert_id then, at least it's clear there is no dedup mechanism for this event, which is ok.

Reply:
Done!